app/assets/javascripts/components/Glossary.js

```diff
@@ -0,0 +1,90 @@
+define(['jquery', 'DoughBaseComponent'], function($, DoughBaseComponent) {
+
+  'use strict';
+
+  var Glossary;
+
+  Glossary = function($el, config) {
+    Glossary.baseConstructor.call(this, $el, config);
+
+    var article = $('[data-dough-glossary-content]'),
+        articleContent = $(article).html(),
+        updatedContent;
+
+    $.getJSON('/glossary.json', function(data){
+      data.glossary.map(function (item) {
+
+      function keywordInContent(keyword, content){
+        return content.indexOf(keyword)>= 1;
+      }
+
+        // If the article content contains any of the keywords in the JSON
+        if(keywordInContent(item.term, articleContent)){
+
+          $(item).each(function(){
+            var regex = "(?!<.*?)\\b(" + item.term + ")\\b(?![^<>]*?(<\/a>|<\/h2>|<\/h3>|>))",
+                unstyledButton = "<button class='unstyled-button glossary__term' aria-label='",
+                sub = "<sub>?</sub></a></button><span class='glossary__description'><span class='glossary__description-title'>",
+                descriptionButton = "<button class='glossary__description-close'><svg xmlns='http://www.w3.org/2000/svg' \
+                class='svg-icon svg-icon--mobile-close-box'><use xmlns:xlink='http://www.w3.org/1999/xlink' xlink:href='#svg-icon--mobile-close-box'>\
+                </use></svg></button></span>"; 
+
+            var updatedContent = articleContent.replace(RegExp(regex,"gi"), unstyledButton + item.term + " (see description)'>" + item.term + sub + item.term + "</a></span>" + item.description + descriptionButton);
```

Comment:
Establishing a line length max is good practice, however... 😄 This component is just to enable an A/B test, which will run on a very small number of pages, then will be removed as its just too much on a intensive process to be going on on the client side.  In the future the terms will be matched on the CMS side and come through to frontend just needing style and click events, so for this instance I think this is ok as it is